### PR TITLE
mobile: removing tracer protos

### PR DESCRIPTION
**Pull request by alyssawilk**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/bazel/api_build_system.bzl [hunk 1]
```diff
@@ -137,7 +137,6 @@ def api_proto_package(
     if srcs == []:
         srcs = native.glob(["*.proto"])
 
-    name = "pkg"
```
Comment (htuch) [inline api/bazel/api_build_system.bzl:140]: Seems like it should have a wider effect than just trace protos, ¯\_(ツ)_/¯

**Review by htuch — COMMENTED** [review]
If this doesn't break stuff it seems reasonable.

File: api/bazel/api_build_system.bzl (hunk 1 above)
Comment (alyssawilk) [reply api/bazel/api_build_system.bzl:140]: given name defaults to pkg above, I suspect there are no non-pkg uses of name today.
I could add a temporary panic if name is non-pkg and check if you're concerned!

Comment (htuch) [reply api/bazel/api_build_system.bzl:140]: No, it's totally fine, if it builds and passes CI, just confused me.

Comment (alyssawilk) [reply api/bazel/api_build_system.bzl:140]: alas fails another part which is hard-coded to strip-prefix pkg.
trying (and failing) to find a workaround so far :-(

**Pull request closed without merging**